pycrdt_websocket/awareness.py

```diff
@@ -1,17 +1,65 @@
 from __future__ import annotations
 
+import asyncio
 import json
 import time
-from typing import Any
+from logging import Logger, getLogger
+from typing import Any, Awaitable, Callable
+from uuid import uuid4
 
-from pycrdt import Decoder, read_message
+import nest_asyncio  # type: ignore[import-untyped]
```

Comment:
We cannot use `nest-asyncio`, why do you need it?

Reply:
Why can't we use it ?

We need it for `loop.run_until_complete(self.set_local_state_field("user", self._user))`, [here](https://github.com/jupyter-server/pycrdt-websocket/pull/73/files#diff-db35ddad107aa5ba618cbd790f840506094190cc99297d1f4aa67a3b7160b395R154).
Otherwise it raises with `event loop already running`.

AFAIU, the `on_change()` callback must be asynchronous to broadcast the message to all users, but the setter must be synchronous.
If we can make [this function](https://github.com/jupyter-server/pycrdt-websocket/pull/73/files#diff-d5f2a35bb64e14b167e41fb7ba849cb6b9cc45197bffb9f11278c457f5848b73R308) synchronous then we can avoid this import.

Reply:
`nest-asyncio` monkey-patches `asyncio`, and has other issues.
Anyway, `pycrdt-websocket` is event-loop agnostic as it's based on AnyIO, so we shouldn't use `asyncio` directly. You can probably spawn a task to solve your issue.

Reply:
> You can probably spawn a task to solve your issue.

Can you be more specific please ? 

Actually this is what I was trying to do with [this function](https://github.com/jupyter-server/pycrdt-websocket/pull/73/files#diff-d5f2a35bb64e14b167e41fb7ba849cb6b9cc45197bffb9f11278c457f5848b73R308), but I still need to call an async function from a sync one.

Reply:
It should be fine actually, thanks.